### ISSUE TITLE
Some `RoutingTest` fails with rack head (Rack 3.1)

### ISSUE THREAD

**Issue by dentarg** [sinatra/sinatra#2014]
From https://github.com/sinatra/sinatra/actions/runs/9313939939/job/25637216301#step:5:72

```
1) Error:
RoutingTest#test_matches_empty_PATH_INFO_to_if_a_route_is_defined_for__0:
Rack::Lint::LintError: PATH_INFO must start with a '/' and must not include a fragment part starting with '#' (origin-form)
    vendor/bundle/ruby/3.2.0/bundler/gems/rack-c8966d083db4/lib/rack/lint.rb:374:in `check_environment'
    vendor/bundle/ruby/3.2.0/bundler/gems/rack-c8966d083db4/lib/rack/lint.rb:63:in `response'
    vendor/bundle/ruby/3.2.0/bundler/gems/rack-c8966d083db4/lib/rack/lint.rb:41:in `call'
    vendor/bundle/ruby/3.2.0/gems/rack-test-2.1.0/lib/rack/test.rb:360:in `process_request'
    vendor/bundle/ruby/3.2.0/gems/rack-test-2.1.0/lib/rack/test.rb:163:in `custom_request'
    vendor/bundle/ruby/3.2.0/gems/rack-test-2.1.0/lib/rack/test.rb:112:in `get'
    /opt/hostedtoolcache/Ruby/3.2.4/x64/lib/ruby/3.2.0/forwardable.rb:240:in `get'
    test/routing_test.rb:273:in `block in <class:RoutingTest>'

  2) Error:
RoutingTest#test_it_handles_encoded_colons_correctly_0:
Rack::Lint::LintError: Only CONNECT requests may have PATH_INFO set to an authority (authority-form)
    vendor/bundle/ruby/3.2.0/bundler/gems/rack-c8966d083db4/lib/rack/lint.rb:364:in `check_environment'
    vendor/bundle/ruby/3.2.0/bundler/gems/rack-c8966d083db4/lib/rack/lint.rb:63:in `response'
    vendor/bundle/ruby/3.2.0/bundler/gems/rack-c8966d083db4/lib/rack/lint.rb:41:in `call'
    vendor/bundle/ruby/3.2.0/gems/rack-test-2.1.0/lib/rack/test.rb:360:in `process_request'
    vendor/bundle/ruby/3.2.0/gems/rack-test-2.1.0/lib/rack/test.rb:163:in `custom_request'
    vendor/bundle/ruby/3.2.0/gems/rack-test-2.1.0/lib/rack/test.rb:112:in `get'
    /opt/hostedtoolcache/Ruby/3.2.4/x64/lib/ruby/3.2.0/forwardable.rb:240:in `get'
    test/routing_test.rb:113:in `block in <class:RoutingTest>'

  3) Error:
RoutingTest#test_matches_empty_PATH_INFO_to_if_no_route_is_defined_for__0:
Rack::Lint::LintError: PATH_INFO must start with a '/' and must not include a fragment part starting with '#' (origin-form)
    vendor/bundle/ruby/3.2.0/bundler/gems/rack-c8966d083db4/lib/rack/lint.rb:374:in `check_environment'
    vendor/bundle/ruby/3.2.0/bundler/gems/rack-c8966d083db4/lib/rack/lint.rb:63:in `response'
    vendor/bundle/ruby/3.2.0/bundler/gems/rack-c8966d083db4/lib/rack/lint.rb:41:in `call'
    vendor/bundle/ruby/3.2.0/gems/rack-test-2.1.0/lib/rack/test.rb:360:in `process_request'
    vendor/bundle/ruby/3.2.0/gems/rack-test-2.1.0/lib/rack/test.rb:163:in `custom_request'
    vendor/bundle/ruby/3.2.0/gems/rack-test-2.1.0/lib/rack/test.rb:112:in `get'
    /opt/hostedtoolcache/Ruby/3.2.4/x64/lib/ruby/3.2.0/forwardable.rb:240:in `get'
    test/routing_test.rb:[25](https://github.com/sinatra/sinatra/actions/runs/9313939939/job/25637216301#step:5:26)5:in `block in <class:RoutingTest>'
```

**Comment by dentarg** [comment]
Looks like this will be addressed in a future Rack 3.1.x release, with https://github.com/rack/rack/pull/2214

No such release yet: https://rubygems.org/gems/rack/versions

**Comment by dentarg** [comment]
It needs to be backported, I opened a PR against the rack 3-1 branch: https://github.com/rack/rack/pull/2234

**Comment by ioquatix** [comment]
This should now be fixed with the latest 3.1 release.
